ratelimits/limiter.go

```diff
@@ -114,21 +127,28 @@ type Decision struct {
 // wait time before the client can make another request, and the time until the
 // bucket refills to its maximum capacity (resets). If no bucket exists for the
 // given limit Name and client id, a new one will be created WITHOUT the
-// request's cost deducted from its initial capacity.
+// request's cost deducted from its initial capacity. If the specified limit is
+// disabled, ErrLimitDisabled is returned.
 func (l *Limiter) Check(ctx context.Context, name Name, id string, cost int64) (*Decision, error) {
 	if cost < 0 {
 		return nil, ErrInvalidCostForCheck
 	}
 
 	limit, err := l.getLimit(name, id)
 	if err != nil {
+		if errors.Is(err, errLimitDisabled) {
+			return disabledLimitDecision, nil
+		}
 		return nil, err
 	}
 
 	if cost > limit.Burst {
 		return nil, ErrInvalidCostOverLimit
 	}
 
+	// Remove cancellation from the request context so that transactions are not
+	// interrupted by a client disconnect.
+	ctx = context.WithoutCancel(ctx)
```

Comment:
As per your comments in source.go:
> Contexts passed to this method should have a timeout or deadline set to prevent the operation from blocking indefinitely.

Removing the outer cancel from these contexts makes sense, but an inner default timeout to prevent indefinite hanging needs to be attached to replace it. Or am I missing a different layer where that separate timeout is added?

Reply:
go-redis is attaching its own deadline to these transactions.

Reply:
Then doesn't that make the comments in source.go inaccurate, since the idea is to rely on a context deadline applied at a lower level?

Reply:
I'm afraid I'm a bit confused. The Redis source relies on go-redis to apply its own context deadline. This is (go-redis) functionality cannot be disabled and uses defaults if none are set in the client. I'm not sure I see the point it applying our own deadline when one already exists.

Reply:
Conversed offline: the comment on each of the `source` interface methods were misleading and implied that contexts passed to them should have a deadline set. I've corrected this now.